<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/ros/project_ws/src/franka_sim/model/robots/robot_door.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="lumi" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <joint name="lumi_joint_base_link" type="fixed">
    <parent link="base_link"/>
    <child link="lumi_link0"/>
    <origin rpy="0 0 0" xyz="-0.4 0.15 0.0"/>
  </joint>
  <link name="lumi_link0">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link0.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link0.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="lumi_link1">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link1.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link1.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="lumi_joint1" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="0 0 0" xyz="0 0 0.333"/>
    <parent link="lumi_link0"/>
    <child link="lumi_link1"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-2.8973" upper="2.8973" velocity="2.1750"/>
  </joint>
  <link name="lumi_link2">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link2.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link2.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="lumi_joint2" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-1.7628" soft_upper_limit="1.7628"/>
    <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
    <parent link="lumi_link1"/>
    <child link="lumi_link2"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-1.7628" upper="1.7628" velocity="2.1750"/>
  </joint>
  <link name="lumi_link3">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link3.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link3.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="lumi_joint3" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="1.57079632679 0 0" xyz="0 -0.316 0"/>
    <parent link="lumi_link2"/>
    <child link="lumi_link3"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-2.8973" upper="2.8973" velocity="2.1750"/>
  </joint>
  <link name="lumi_link4">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link4.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link4.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="lumi_joint4" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-3.0718" soft_upper_limit="-0.0698"/>
    <origin rpy="1.57079632679 0 0" xyz="0.0825 0 0"/>
    <parent link="lumi_link3"/>
    <child link="lumi_link4"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-3.0718" upper="-0.0698" velocity="2.1750"/>
  </joint>
  <link name="lumi_link5">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link5.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link5.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="lumi_joint5" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="-1.57079632679 0 0" xyz="-0.0825 0.384 0"/>
    <parent link="lumi_link4"/>
    <child link="lumi_link5"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-2.8973" upper="2.8973" velocity="2.6100"/>
  </joint>
  <link name="lumi_link6">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link6.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link6.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="lumi_joint6" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-0.0175" soft_upper_limit="3.7525"/>
    <origin rpy="1.57079632679 0 0" xyz="0 0 0"/>
    <parent link="lumi_link5"/>
    <child link="lumi_link6"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-0.0175" upper="3.7525" velocity="2.6100"/>
  </joint>
  <link name="lumi_link7">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link7.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/link7.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="lumi_joint7" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="1.57079632679 0 0" xyz="0.088 0 0"/>
    <parent link="lumi_link6"/>
    <child link="lumi_link7"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-2.8973" upper="2.8973" velocity="2.6100"/>
  </joint>
  <link name="lumi_link8"/>
  <joint name="lumi_joint8" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.107"/>
    <parent link="lumi_link7"/>
    <child link="lumi_link8"/>
    <axis xyz="0 0 0"/>
  </joint>
  <joint name="lumi_hand_joint" type="fixed">
    <parent link="lumi_link8"/>
    <child link="lumi_hand"/>
    <origin rpy="0 0 -0.785398163397" xyz="0 0 0"/>
  </joint>
  <link name="lumi_hand">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/hand.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/hand.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="lumi_leftfinger">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/finger.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/finger.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="lumi_rightfinger">
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/finger.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/collision/finger.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="lumi_finger_joint1" type="prismatic">
    <parent link="lumi_hand"/>
    <child link="lumi_leftfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 1 0"/>
    <limit effort="20" lower="0.0" upper="0.04" velocity="0.2"/>
  </joint>
  <joint name="lumi_finger_joint2" type="prismatic">
    <parent link="lumi_hand"/>
    <child link="lumi_rightfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 -1 0"/>
    <limit effort="20" lower="0.0" upper="0.04" velocity="0.2"/>
    <mimic joint="lumi_finger_joint1"/>
  </joint>
  <link name="lumi_ee"/>
  <joint name="ee_fjoint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.116"/>
    <parent link="lumi_hand"/>
    <child link="lumi_ee"/>
  </joint>
  <mujoco>
    <compiler balanceinertia="true" discardvisual="false" meshdir="../meshes/"/>
  </mujoco>
  <link name="base_link"/>
  <link name="door_base"/>
  <joint name="door_base_joint" type="fixed">
    <origin rpy="0 0 -1.57079632679" xyz="0.4 0.25 0.052"/>
    <parent link="base_link"/>
    <child link="door_base"/>
  </joint>
  <link name="door_link"/>
  <joint name="door_hinge" type="revolute">
    <origin rpy="0 0 0" xyz="0.262 0.049 0"/>
    <limit effort="10" lower="0" upper="1.57079632679" velocity="10"/>
    <axis xyz="0 0 -1"/>
    <parent link="door_base"/>
    <child link="door_link"/>
  </joint>
  <link name="door">
    <visual>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/door_v5_bin.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/door_v5_bin.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="door_to_hinge" type="fixed">
    <origin rpy="0 0 0" xyz="-0.257 -0.044 0.005"/>
    <parent link="door_link"/>
    <child link="door"/>
  </joint>
  <link name="handle_core">
    <visual>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/handle_v5_core_bin.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/handle_v5_core_bin.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="handle_core" type="revolute">
    <origin rpy="0 0 0" xyz="0.052 -0.06 0.095"/>
    <limit effort="10" lower="0" upper="0.523598775598" velocity="10"/>
    <axis xyz="0 1 0"/>
    <parent link="door"/>
    <child link="handle_core"/>
  </joint>
  <!-- handle_beam collision mesh replaced by dummy due to issues with moveit collisions between robot left/right fingers and handle_beam -->
  <link name="handle_beam">
    <visual>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/handle_v5_beam_bin.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <box size="0.005 0.005 0.005"/>
      </geometry>
    </collision>
    <!--<collision>
           <geometry>
               <mesh filename="package://franka_sim/model/meshes/handle_v5_beam_bin.stl"/>
           </geometry>
        </collision>-->
  </link>
  <joint name="handle_beam_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.015 0.0 -0.0165"/>
    <parent link="handle_core"/>
    <child link="handle_beam"/>
  </joint>
  <link name="handle_latch">
    <visual>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/handle_v5_latch_bin.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/handle_v5_latch_bin.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="handle_latch_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.015 0.04 -0.01"/>
    <parent link="handle_core"/>
    <child link="handle_latch"/>
  </joint>
  <link name="frame_left">
    <visual>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/frame_v5_leftpillar_back_bin.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/frame_v5_leftpillar_back_bin.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="frame_left_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.025 0.01 0.0"/>
    <parent link="door_base"/>
    <child link="frame_left"/>
  </joint>
  <link name="frame_left_front">
    <visual>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/frame_v5_leftpillar_front_bin.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/frame_v5_leftpillar_front_bin.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="frame_left_front_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.025 0.0 0.0"/>
    <parent link="door_base"/>
    <child link="frame_left_front"/>
  </joint>
  <link name="frame_right">
    <visual>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/frame_v5_rightpillar_bin.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/frame_v5_rightpillar_bin.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="frame_right_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.267 0.0 0.0"/>
    <parent link="door_base"/>
    <child link="frame_right"/>
  </joint>
  <link name="frame_bottom">
    <visual>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/frame_v5_top_bottom_bin.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/frame_v5_top_bottom_bin.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="frame_bottom_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.025 0.0 -0.052"/>
    <parent link="door_base"/>
    <child link="frame_bottom"/>
  </joint>
  <link name="frame_top">
    <visual>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/frame_v5_top_bottom_bin.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://franka_sim/model/meshes/frame_v5_top_bottom_bin.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="frame_top_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.025 0.0 0.335"/>
    <parent link="door_base"/>
    <child link="frame_top"/>
  </joint>
</robot>
<!-- <asset>
        <mesh file="stl/frame_v5_leftpillar_back_bin.stl" name="frame_left"/>
        <mesh file="stl/frame_v5_leftpillar_front_bin.stl" name="frame_left_front"/>
        <mesh file="stl/frame_v5_rightpillar_bin.stl" name="frame_right"/>
        <mesh file="stl/frame_v5_top_bottom_bin.stl" name="frame_top_bottom"/>
        <mesh file="stl/door_v5_bin.stl" name="door"/>
        <mesh file="stl/handle_v5_beam_bin.stl" name="handle_beam"/>
        <mesh file="stl/handle_v5_latch_bin.stl" name="handle_latch"/>
        <mesh file="stl/handle_v5_core_bin.stl" name="handle_core"/>
</asset> -->
<!-- <body pos="0 0 0">
    <body name="door" pos="5 5 5">
        <body name="handle_core" pos="52 -60 95">
            <body name="handle_beam" pos="15 0 -16.5">
            </body>
            <body name="handle_latch" pos="-15 40 -10">
            </body>
        </body>    
    </body>          
    <body name="frame_left" pos="-25 10 0">
    </body>
    <body name="frame_left_front" pos="-25 0 0">
    </body>
    <body name="frame_right" pos="267 0 0">
    </body>
    <body name="frame_bottom" pos="-25 0 -52">
    </body>
    <body name="frame_top" pos="-25 0 335">
    </body>
</body>    -->
